File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Float}
def find_median_sorted_arrays(nums1, nums2)
    i=0
    j=0
    while i < nums1.length
        break if j >= nums2.length
        while i < nums1.length && nums2[j] > nums1[i]
            i+=1
        end
        nums1.insert(i,nums2[j])
        j+=1
        i+=1
    end
    nums1.insert(-1, *nums2[j, nums2.length]) if j < nums2.length
    # p nums1
    if (nums1.length%2)!=0
        return nums1[nums1.length/2]
    else
        p nums1.length/2
        return (nums1[nums1.length/2]+nums1[(nums1.length-2)/2]).to_f/2
    end
end

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    let merged=[]
    let i=0,j=0
    while(i<nums1.length){
        while(j<nums2.length&&nums2[j]<=nums1[i]){
            merged.push(nums2[j])
            j+=1
        }
        merged.push(nums1[i])
        i+=1
    }
    merged.push(...nums2.slice(j, nums2.length))
    // console.log(merged)
    let len=merged.length-1
    return (merged[Math.floor(len/2)]+merged[Math.ceil(len/2)])/2
};